node/src/accountant/scanners/payable_scanner/tx_templates/initial/new.rs

```diff
@@ -0,0 +1,223 @@
+
+        assert_eq!(templates.len(), 2);
+        assert_eq!(templates[0], template1);
+        assert_eq!(templates[1], template2);
+        assert!(!templates.is_empty());
+        assert!(templates.contains(&template1));
+        assert_eq!(
+            templates
+                .iter()
+                .map(|template| template.base.amount_in_wei)
+                .sum::<u128>(),
+            3000
+        );
+    }
+
+    #[test]
+    fn new_tx_templates_into_iter_consumes_and_iterates() {
+        let template1 = NewTxTemplate {
+            base: BaseTxTemplate {
+                receiver_address: make_address(1),
+                amount_in_wei: 1000,
+            },
+        };
+        let template2 = NewTxTemplate {
+            base: BaseTxTemplate {
+                receiver_address: make_address(2),
+                amount_in_wei: 2000,
+            },
+        };
+        let templates = NewTxTemplates(vec![template1.clone(), template2.clone()]);
+
+        let collected: Vec<NewTxTemplate> = templates.into_iter().collect();
+
+        assert_eq!(collected.len(), 2);
+        assert_eq!(collected[0], template1);
+        assert_eq!(collected[1], template2);
+    }
+
+    #[test]
+    fn new_tx_templates_can_be_created_from_payable_accounts() {
+        let wallet1 = make_wallet("wallet1");
+        let wallet2 = make_wallet("wallet2");
+        let payable_accounts = vec![
+            PayableAccount {
+                wallet: wallet1.clone(),
+                balance_wei: 1000,
+                last_paid_timestamp: SystemTime::now(),
+                pending_payable_opt: None,
+            },
+            PayableAccount {
+                wallet: wallet2.clone(),
+                balance_wei: 2000,
+                last_paid_timestamp: SystemTime::now(),
+                pending_payable_opt: None,
+            },
+        ];
+
+        let new_tx_templates = NewTxTemplates::from(&payable_accounts);
+
+        assert_eq!(new_tx_templates.len(), 2);
+        assert_eq!(new_tx_templates[0].base.receiver_address, wallet1.address());
+        assert_eq!(new_tx_templates[0].base.amount_in_wei, 1000);
+        assert_eq!(new_tx_templates[1].base.receiver_address, wallet2.address());
+        assert_eq!(new_tx_templates[1].base.amount_in_wei, 2000);
+    }
+
+    #[test]
+    fn new_tx_templates_can_be_created_from_iterator() {
+        let template1 = NewTxTemplate {
+            base: BaseTxTemplate {
+                receiver_address: make_address(1),
+                amount_in_wei: 1000,
+            },
+        };
+        let template2 = NewTxTemplate {
+            base: BaseTxTemplate {
+                receiver_address: make_address(2),
+                amount_in_wei: 2000,
+            },
+        };
+
+        let templates_vec = vec![template1.clone(), template2.clone()];
+        let templates = NewTxTemplates::from_iter(templates_vec.into_iter());
```

Comment:
This looks completely redundant. Cannot you simply construct it like NewTxTemplates::new(vec![template1.clone(), template2.clone()]) or do you really need the polymorphism and apply it to different collections??

Maybe you can splash the IntoIterator impl into the toilet.